cmd/soroban-cli/src/strval.rs

```diff
@@ -909,24 +929,43 @@ pub fn to_json(v: &ScVal) -> Result<Value, Error> {
             .to_string();
             Value::String(v)
         }
-        ScVal::U256(Uint256(inner)) => {
-            let (hi, lo) = inner.split_at(16);
-            Value::String(
-                ethnum::U256::from_words(
-                    u128::from_be_bytes(hi.try_into()?),
-                    u128::from_be_bytes(lo.try_into()?),
-                )
-                .to_string(),
-            )
+        ScVal::U256(u256parts) => {
+            let hi_bytes = [u256parts.hi_hi.to_be_bytes(), u256parts.hi_lo.to_be_bytes()].concat();
+            let lo_bytes = [u256parts.lo_hi.to_be_bytes(), u256parts.lo_lo.to_be_bytes()].concat();
+            let hi = u128::from_be_bytes(
+                hi_bytes
+                    .as_slice()
+                    .try_into()
+                    .map_err(|_| Error::InvalidValue(Some(ScType::U256)))?,
+            );
+            let lo = u128::from_be_bytes(
+                lo_bytes
+                    .as_slice()
+                    .try_into()
+                    .map_err(|_| Error::InvalidValue(Some(ScType::U256)))?,
+            );
+            Value::String(ethnum::U256::from_words(hi, lo).to_string())
         }
-        ScVal::I256(Uint256(inner)) => {
-            let (hi, lo) = inner.split_at(16);
+        ScVal::I256(i256parts) => {
+            let hi_bytes = [i256parts.hi_hi.to_be_bytes(), i256parts.hi_lo.to_be_bytes()].concat();
+            let lo_bytes = [i256parts.lo_hi.to_be_bytes(), i256parts.lo_lo.to_be_bytes()].concat();
+            let hi = i128::from_be_bytes(
+                hi_bytes
+                    .as_slice()
+                    .try_into()
+                    .map_err(|_| Error::InvalidValue(Some(ScType::I256)))?,
+            );
+
+            let lo = u128::from_be_bytes(
+                lo_bytes
+                    .as_slice()
+                    .try_into()
+                    .map_err(|_| Error::InvalidValue(Some(ScType::I256)))?,
+            );
             Value::String(
-                ethnum::I256::from_words(
-                    i128::from_be_bytes(hi.try_into()?),
-                    i128::from_be_bytes(lo.try_into()?),
-                )
-                .to_string(),
+                // TODO: double-check that the 'as' below is correct
+                #[allow(clippy::cast_possible_wrap)]
+                ethnum::I256::from_words(hi, lo as i128).to_string(),
```

Comment:
Wouldn't it be easier here to use `I256::from_be_bytes`, instead of building words?
```rs
let bytes: [u8; 32] = [
  i256parts.hi_hi.to_be_bytes(),
  i256parts.hi_lo.to_be_bytes(),
  i256parts.lo_hi.to_be_bytes(),
  i256parts.lo_lo.to_be_bytes(),
].concat();
Value::String(
  ethnum::I256::from_be_bytes(
    // even unwrap here might be fine, as we know there are 32 bytes
    bytes.as_slice().try_into()?
  ).to_string(),
)
```